### PR TITLE
Hibernate ucp sample

### PR DESCRIPTION
**Pull request by sreekarareddy**
Adds a new sample hibernate project that shows a demo of how to configure UCP in a standalone hibernate application through config/properties file. This configuration is achieved using the "hibernate-ucp" module added in Hibernate version 6.5.

### CODE REVIEW

File: java/hibernate-ucp/src/main/resources/hibernate.cfg.xml
Comment (jeandelavarene) [inline java/hibernate-ucp/src/main/resources/hibernate.cfg.xml:1]: This xml config file isn't used, right? If it's not used, I think it's best to not include it.

File: java/hibernate-ucp/src/main/resources/hibernate.properties
Comment (jeandelavarene) [inline java/hibernate-ucp/src/main/resources/hibernate.properties:1]: Is this file also used? Or just the xml config one?

File: java/hibernate-ucp/src/main/resources/hibernate.cfg.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        "-//Hibernate/Hibernate Configuration DTD 3.0//EN" 
+        "http://www.hibernate.org/dtd/hibernate-configuration-3.0.dtd">
+
+<hibernate-configuration>
+    <session-factory>
+        <property name="hbm2ddl.auto">create</property>
+        <property name="connection.url">url</property>
+        <property name="connection.username">user</property>
+        <property name="connection.password">password</property>
+        <property name="connection.driver_class">oracle.jdbc.driver.OracleDriver</property>
```
Comment (jeandelavarene) [inline java/hibernate-ucp/src/main/resources/hibernate.cfg.xml:12]: Why do you have both connection.driver_class and oracleucp.connectionFactoryClassName? It looks redundant.

File: java/hibernate-ucp/src/main/resources/hibernate.properties
Comment (sreekarareddy) [reply java/hibernate-ucp/src/main/resources/hibernate.properties:1]: Both the files are used. hibernate.properties file is loaded by default

File: java/hibernate-ucp/src/main/resources/hibernate.cfg.xml (hunk 1 above)
Comment (sreekarareddy) [reply java/hibernate-ucp/src/main/resources/hibernate.cfg.xml:12]: driver_class is mandatory for hibernate

**Review by jeandelavarene — APPROVED** [review]

**Pull request merged**